### ISSUE TITLE
support wait/nowait grammar on ddl operation

### ISSUE THREAD

**Issue by Damon-PingCap** [pingcap/tidb#19928]
## Feature Request

**Is your feature request related to a problem? Please describe:**

加索引时在终端被 kill 或者 Ctrl+c 之后与 MySQL 行为不一致，造成混乱。并且一次加多个索引通常需要开多个 session。
另外有些 ddl 也可能被加索引操作长时间阻塞。
比如一个正在加索引的表上的 add column 操作会被阻塞，同时又会阻塞其它表的 ddl 操作，比如 create table ，drop table 之类的。均需要能放到后台的方法，当前是通过 Ctrl+c 来达到效果，与正常的理解有偏差。

**Describe the feature you'd like:**

1，在执行长时间的加索引操作时，可以通过 nowait 选项，让前台 session 返回，添加索引操作自动在后台运行，前台不需要等待。
2，ddl 操作在被 kill ，或者手动 Ctrl+c 断开之后，取消当前执行的 DDL ，保持正常的理解逻辑。
3，当前 admin show ddl 比较难分辨阻塞关系，需要处于 wait 状态的 ddl 操作能直观的看出来被哪个 job 阻塞。

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by nullnotnil** [comment]
MariaDB [currently supports this syntax on DDL](https://mariadb.com/kb/en/wait-and-nowait/).

The use case is different though: MySQL/MariaDB will not permit schema changes while there is an open transaction that has modified data on the table. See https://github.com/pingcap/tidb/issues/17796 for an example. I don't mind TiDB's behavior here, but it could be better documented.

w.r.t. the use case in TiDB. My hope is that https://github.com/pingcap/tidb/issues/18396 is implemented instead. But if the scope makes it too large of a feature, this serves a useful intermediate use-case.

**Comment by bb7133** [comment]
Thanks for the tips @nullnotnil . considering the different semantic of `WAIT`/`NOTWAIT` between this proposal and MariaDB, I think we can use `ASYNC` keyword instead?

**Comment by nullnotnil** [comment]
@bb7133 yes I agree, `ASYNC` better describes the semantic requested. Another sql-like word is `deferred`, borrowed from the feature [deferrable constraints](https://begriffs.com/posts/2017-08-27-deferrable-sql-constraints.html) (not available in mysql)

There might be some hidden scope for reviewers to discuss:

The feature request says that `admin show ddl` has intuitiveness issues, but an `async` ddl job that has to be killed because it failed to convert a type without data loss could be harder to debug. There is the opportunity to support only a subset of 'safe' DDL statements, but that also could create issues for documentation to clarify what is supported for each type of DDL.
